src/main/java/com/example/git_repo/error/decoder/GitErrorDecoder.java

```diff
@@ -0,0 +1,18 @@
+package com.example.git_repo.error.decoder;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+public class GitErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        if (response.status() == HttpStatus.SERVICE_UNAVAILABLE.value()) {
+            return new ResponseStatusException(HttpStatus.SERVICE_UNAVAILABLE, "Service Unavailable - 503");
```

Comment:
Nie spowoduje to wykonania retry, w wymaganiach zadnaia bylo aby w sytuacji kiedy zwrocony z jakiejs aplikacji zostanie status 503 to powinien wykonac sie retry